src/core/Keymap.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Keymap.cpp - implementation of keymap class
+ *
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "Keymap.h"
+
+#include <QDomElement>
+
+
+Keymap::Keymap() :
+	m_description(tr("empty")),
+	m_firstKey(0),
+	m_lastKey(127),
+	m_middleKey(DefaultMiddleKey),
+	m_baseKey(DefaultBaseKey),
+	m_baseFreq(DefaultBaseFreq)
+{
+}
+
+
+Keymap::Keymap(	QString description,
+				std::vector<int> newMap,
+				int newFirst,
+				int newLast,
+				int newMiddle,
+				int newBaseKey,
+				float newBaseFreq) :
```

Comment:
No, just alignment messed up by the fact that github forces 8-spaces sized tabs. :) (In this case it seemed better to add a tab before `QString description` instead of adding one or two spaces to all other lines to align them.)

Reply:
If you want them aligned, use spaces. Tabs for indent + spaces for allignment allows people to view at their preferred tab size w/o anything breaking.

Reply:
Our `.editorconfig` should tell the github view to use `4`-sized-tabs. But github's support is inofficial and sometimes, it does not work. :disappointed:

Reply:
I heard about the "indent + spaces" idea before, but it seems evil to me to mix tabs and spaces, even for such a specific case. Also, I thought we picked and standardized the 4-spaces wide tabs exactly to make sure the code stays consistent for everyone. If the code should be written to look always good regardless the size of tabs, we should have picked a "use tabs, whatever size you want" rule, because the size would make no difference..

But anyway, in this case it would be probably easier and less controversial to just put the `QString description` to a new line and just indent by one tab..